components/blocks/Recipe.tsx

```diff
@@ -0,0 +1,320 @@
+                      {...props}
+                      highlightLines={highlightLines}
+                    />
+                  ),
+                }}
+              />
+            </div>
+          ) : (
+            <p>No code block available.</p>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default RecipeBlock;
+
+interface CodeBlockProps {
+  value: any;
+  lang?: string;
+  children?: React.ReactNode;
+  highlightLines: string;
+}
+
+export const CodeToolbar = ({
+  lang,
+  onCopy,
+  tooltipVisible,
+}: {
+  lang?: string;
+  onCopy: () => void;
+  tooltipVisible: boolean;
+}) => (
+  <div className="code-toolbar bg-gray-800 text-white px-4 py-2 lg:rounded-t-xl text-sm font-semibold flex justify-between items-center">
+    <span className="font-tuner">{lang || 'Unknown'}</span>
+    <div className="flex items-center ml-4 space-x-4 relative overflow-visible">
+      <button
+        onClick={onCopy}
+        className={`flex items-center px-2 py-1 bg-gray-800  rounded-md text-sm transition-colors duration-200 space-x-1 relative ${
+          tooltipVisible
+            ? 'text-white bg-gray-700 rounded-md ml-1'
+            : 'hover:bg-gray-700 text-white'
+        }`}
+      >
+        {!tooltipVisible && <MdOutlineContentCopy className="w-4 h-4" />}
+        <span>{!tooltipVisible ? 'Copy' : 'Copied!'}</span>
+      </button>
+    </div>
+  </div>
+);
+
+const CodeBlockWithHighlightLines = ({
+  value,
+  lang,
+  children,
+  highlightLines,
+}: CodeBlockProps) => {
+  const [isMounted, setIsMounted] = useState(false);
+  const [tooltipVisible, setTooltipVisible] = useState(false);
+
+  useEffect(() => {
+    setIsMounted(true);
+    if (isMounted) {
+      Prism.highlightAll();
+    }
+  }, [isMounted]);
+
+  useEffect(() => {
+    Prism.highlightAll();
+  }, [highlightLines]);
+
+  if (!isMounted) {
+    return null;
+  }
+
+  const copyToClipboard = () => {
+    const codeToCopy = typeof children === 'string' ? children : value;
+    navigator.clipboard.writeText(codeToCopy).then(
+      () => {
+        setTooltipVisible(true);
+        setTimeout(() => setTooltipVisible(false), 1500);
+      },
+      (err) => {
+        console.error('Failed to copy code:', err);
+      }
+    );
+  };
+
+  return (
+    <div className="codeblock-container">
+      <div className="sticky top-0 z-30">
+        <CodeToolbar
+          lang={lang}
+          onCopy={copyToClipboard}
+          tooltipVisible={tooltipVisible}
+        />
+      </div>
+      <pre
+        className="line-numbers"
+        data-line={highlightLines}
+        style={{
+          overflowX: 'hidden',
+          maxWidth: '100%',
+          whiteSpace: 'pre-wrap',
+          wordBreak: 'break-word',
+        }}
+      >
+        <code className={`language-${lang || 'jsx'}`}>
+          {typeof children === 'string' || children ? children : value}
+        </code>
+      </pre>
+    </div>
+  );
+};
+
+const customHighlightCSS = `
```

Comment:
1. Add comment - need this CSS to edit Prism themes
2. Move to another file - its yucky

Reply:
Done vboth